sake/redis.py

```diff
@@ -1203,102 +1427,99 @@ def index(cls) -> typing.Sequence[ResourceIndex]:
 
-                try:
-                    user = await self.rest.rest.fetch_my_user()
+        else:
+            self.metadata.pop("chunk_on_create", None)
 
-                except hikari_errors.RateLimitedError as exc:
-                    back_off.set_next_backoff(exc.retry_after)
+        return self
 
-                except hikari_errors.InternalServerError:
-                    pass
+    async def __bulk_set_members(self, members: typing.Iterator[ObjectT], /, guild_id: int) -> None:
+        client = self.get_connection(ResourceIndex.MEMBER)
+        windows = redis_iterators.chunk_values((_add_guild_id(payload, guild_id) for payload in members))
+        setters = (client.hmset_dict(guild_id, _get_sub_user_id, window) for window in windows)
+        user_setter = self._try_bulk_set_users(member["user"] for member in members)
+        await asyncio.gather(*setters, user_setter)
+
+    @utility.as_raw_listener("GUILD_CREATE")  # members aren't included in GUILD_UPDATE
+    async def __on_guild_create(self, event: shard_events.ShardPayloadEvent, /) -> None:
+        guild_id = int(event.payload["id"])
+        await self.clear_members_for_guild(guild_id)
+        await self.__bulk_set_members(event.payload["members"], guild_id)
+
+        if shard_aware := self.metadata.get("chunk_on_create"):
+            assert isinstance(shard_aware, hikari_traits.ShardAware)
+            include_presences = bool(self.metadata.get("chunk_presences", False))
+            await shard_aware.request_guild_members(guild_id, include_presences=include_presences)
+
+    @utility.as_listener(guild_events.GuildLeaveEvent)
+    async def __on_guild_leave_event(self, event: guild_events.GuildLeaveEvent, /) -> None:
+        await self.clear_members_for_guild(event.guild_id)
 
-                else:
-                    self.metadata["own_id"] = user.id
-                    break
+    @utility.as_raw_listener("GUILD_MEMBER_ADD", "GUILD_MEMBER_UPDATE")
+    async def __on_guild_member_add_update(self, event: shard_events.ShardPayloadEvent, /) -> None:
+        await self.set_member(dict(event.payload), int(event.payload["guild_id"]))
 
-            own_id = snowflakes.Snowflake(self.metadata["own_id"])
-            if event.user_id == own_id:
-                await self.clear_members_for_guild(event.guild_id)
-            else:
-                await self.delete_member(event.guild_id, event.user_id)
+    @utility.as_listener(member_events.MemberDeleteEvent)
+    async def __on_member_delete_event(self, event: member_events.MemberDeleteEvent, /) -> None:
+        try:
+            own_id = self.metadata[_OwnIDStore.KEY]
+            assert isinstance(own_id, _OwnIDStore)
 
-    async def __on_member_chunk_event(self, event: shard_events.MemberChunkEvent) -> None:
-        await self.__bulk_add_members(event.guild_id, event.members)
+        except KeyError:
+            own_id = self.metadata[_OwnIDStore.KEY] = _OwnIDStore(self.rest)
```

Comment:
We need to remove the _OwnIDStore on close to ensure no asyncio stuff is preserved between loops or whatever